learn/functions/recursion.md

```diff
@@ -0,0 +1,124 @@
+**Recursive programming** is a technique in programming where a function **calls itself repeatedly** until it reaches a base or terminal case. It is a powerful tool when dealing with certain types of problems that can be naturally defined in a recursive manner. In Python, we can implement this technique through **recursive functions.**
```

Comment:
`**recursive functions.**` bold wrapper shouldn't include the dot `.` of the sentence.